debug/graphviz.c

```diff
@@ -0,0 +1,907 @@
+/**
+ * @file
+ * Create a GraphViz dot file from the NeoMutt objects
+ *
+ * @authors
+ * Copyright (C) 2018-2020 Richard Russon <rich@flatcap.org>
+ *
+ * @copyright
+ * This program is free software: you can redistribute it and/or modify it under
+ * the terms of the GNU General Public License as published by the Free Software
+ * Foundation, either version 2 of the License, or (at your option) any later
+ * version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/**
+ * @page debug_graphviz Create a GraphViz dot file
+ *
+ * Create a GraphViz dot file from the NeoMutt objects
+ */
+
+#include "config.h"
+#include <stdio.h>
+#include <string.h>
+#include <sys/stat.h>
+#include <time.h>
+#include "imap/imap_private.h"
+#include "maildir/maildir_private.h"
+#include "notmuch/notmuch_private.h"
+#include "pop/pop_private.h"
+#include "email/lib.h"
+#include "core/lib.h"
+#include "conn/lib.h"
+#include "context.h"
+#include "globals.h"
+#include "compress/lib.h"
+#include "mbox/lib.h"
+#include "nntp/lib.h"
+#include "notmuch/lib.h"
+
+// #define GV_HIDE_CONTEXT
+#define GV_HIDE_CONTEXT_CONTENTS
+// #define GV_HIDE_MBOX
+// #define GV_HIDE_NEOMUTT
+// #define GV_HIDE_CONFIG
+// #define GV_HIDE_MDATA
+
+void dot_type_bool(FILE *fp, const char *name, bool val)
```

Comment:
Can we please make all internal functions static?

Reply:
Done